### PR TITLE
Add transaction status to blocks_internal_commands

### PR DESCRIPTION
**Pull request by deepthiskumar**
Building on #11331
Explain your changes:
* Rename user_command_status to transaction_status
* Add status and failure reason to `blocks_internal_commands` table
* Update extensional blocks

Explain how you tested your changes:
* Ran daemon and archive node locally
* Updated the replayer test database 

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

File: src/app/archive/archive_lib/processor.ml [hunk 1]
```diff
@@ -2781,19 +2854,63 @@ module Block = struct
...
                       in
+                      let status, failure_reason =
+                        match status with
+                        | Applied ->
+                            (applied_str, None)
+                        | Failed failures -> (
+                            (* at most two failures in a coinbase transaction First one for the fee transfer and the second for reward transfer*)
+                            match failures with
+                            | [ []; _ ] ->
+                                applied_status
```
Comment (psteckler) [inline src/app/archive/archive_lib/processor.ml:2865]: should this case ever happen? is it an internal error if it does happen?

File: src/app/archive/archive_lib/processor.ml [hunk 2]
```diff
@@ -2781,19 +2854,63 @@ module Block = struct
...
+                            | [ [ failure1 ]; _ ] ->
+                                (failed_str, Some failure1)
+                            | _ ->
```
Comment (psteckler) [inline src/app/archive/archive_lib/processor.ml:2868]: is the 2-failure case missing?

File: src/app/archive/archive_lib/processor.ml [hunk 3]
```diff
@@ -2781,19 +2854,63 @@ module Block = struct
...
+                let status, failure_reason =
+                  match status with
+                  | Applied ->
+                      (applied_str, None)
+                  | Failed failures -> (
+                      (* at most two failures in a coinbase transaction First one for the fee transfer (if any) and the second for reward transfer*)
+                      match
+                        (failures, Option.is_none fee_transfer_via_coinbase)
+                      with
+                      | [ [] ], true ->
```
Comment (psteckler) [inline src/app/archive/archive_lib/processor.ml:2893]: internal error?

File: src/app/archive/archive_lib/processor.ml (hunk 1 above)
Comment (deepthiskumar) [reply src/app/archive/archive_lib/processor.ml:2865]: This is the case where the coinbase part of the transaction could fail but the fee transfer succeeds (the coinbase transaction returns as failed) and we are retrieving only the status of the fee tansfer here

File: src/app/archive/archive_lib/processor.ml (hunk 2 above)
Comment (deepthiskumar) [reply src/app/archive/archive_lib/processor.ml:2868]: Failure for coinbase is obtained below

Comment (deepthiskumar) [reply src/app/archive/archive_lib/processor.ml:2889]: Added comments about the structure in the code

**Review by MartinMinkov — APPROVED** [review]

**Review by psteckler — APPROVED** [review]

**Pull request merged**